core/benches/bench_hashmap.rs

```diff
@@ -0,0 +1,302 @@
+        b.iter_batched_ref(
+            || setup_hashmap(),
+            |hmap| populated_cache::insert_and_inc(hmap),
+            BatchSize::SmallInput,
+        )
+    });
+    group.bench_function("insert_and_inc_entry_api", |b| {
+        b.iter_batched_ref(
+            || setup_hashmap(),
+            |hmap| populated_cache::insert_and_inc_entry_api(hmap),
+            BatchSize::SmallInput,
+        )
+    });
+    group.finish();
+}
+
+fn bench_remove_populated_cache(c: &mut Criterion) {
+    let _ = env::test::run_test::<env::DefaultEnvTypes, _>(|_| {
+        let mut group = c.benchmark_group(
+            "Compare: `remove` and `remove_entry_api` (populated cache)",
+        );
+        group.bench_function("remove", |b| {
+            b.iter_batched_ref(
+                || setup_hashmap(),
+                |hmap| populated_cache::remove(hmap),
+                BatchSize::SmallInput,
+            )
+        });
+        group.bench_function("remove_entry_api", |b| {
+            b.iter_batched_ref(
+                || setup_hashmap(),
+                |hmap| populated_cache::remove_entry_api(hmap),
+                BatchSize::SmallInput,
+            )
+        });
+        group.finish();
+        Ok(())
+    })
+    .unwrap();
+}
+
+mod empty_cache {
+    use super::*;
+
+    /// Iteratively checks if an entry is in the `StorageHashMap`. If not, it
+    /// is inserted. In either case it is incremented afterwards.
+    pub fn insert_and_inc(hmap: &mut StorageHashMap<i32, i32>) {
+        for key in 0..ENTRIES * 2 {
+            black_box({
+                if !hmap.contains_key(&key) {
+                    hmap.insert(key, key);
+                }
+                *hmap.get_mut(&key).unwrap() += 1;
+            });
+        }
+    }
+
+    /// Iteratively checks if an entry is in the `StorageHashMap`. If not, it
+    /// is inserted. In either case it is incremented afterwards.
+    pub fn insert_and_inc_entry_api(hmap: &mut StorageHashMap<i32, i32>) {
+        for key in 0..ENTRIES * 2 {
+            black_box({
+                let v = hmap.entry(key).or_insert(key);
+                *v += 1;
+            });
+        }
+    }
+
+    /// Iteratively checks if an entry is in the `StorageHashMap`. If yes, it
+    /// is taken out.
+    pub fn remove(hmap: &mut StorageHashMap<i32, i32>) {
+        for key in 0..ENTRIES {
+            black_box({
+                if hmap.contains_key(&key) {
+                    let _ = hmap.take(&key);
+                }
+            });
+        }
+    }
+
+    /// Iteratively checks if an entry is in the `StorageHashMap`. If yes, it
+    /// is taken out.
+    ///
+    /// Uses the Entry API.
+    pub fn remove_entry_api(hmap: &mut StorageHashMap<i32, i32>) {
+        for key in 0..ENTRIES {
+            black_box({
+                if let Entry::Occupied(o) = hmap.entry(key) {
+                    o.remove();
+                }
+            });
+        }
+    }
+}
```

Comment:
Correct me if I am wrong but this looks like code dupe from populated cache. We could just use the same methods for both benchmarks but with different setups, right?